examples/nextjs-route-handlers-upload/README.md

```diff
@@ -1,3 +1,99 @@
-# Uploading Files to Cloudinary with Next.js Route Handlers
+# Uploading Files to Cloudinary with Next.js API Routes/Route Handlers
 
-Coming soon
\ No newline at end of file
+This example shows how to upload files to Cloudinary using Next.js API Routes (Pages Router) and Route Handlers (App Router).
+
+## Getting Started
+
+### Using the pages router
+
+Create an API Route as shown in [upload.ts](src/pages/api/upload.ts) using the `formidable` library to accept files in request body.
+
+```ts
+const uploadsFolder = process.env.CLOUDINARY_UPLOADS_FOLDER;
+const form = new IncomingForm();
+form.parse(req, async (err, fields, files) => {
+  const result = await new Promise<UploadApiResponse>((resolve, reject) => {
+    cloudinary.uploader.upload(
+      file.filepath,
+      { resource_type: 'auto', folder: uploadsFolder },
+      (error, result) => {
+        if (error || !result) reject(error);
+        else resolve(result);
+      },
+    );
+});
+```
+
+From the client side, you can use the `fetch` API to send a POST request to the API Route with the file in the request body as shown in [index.tsx](src/pages/index.tsx).
+
+```tsx
+// handleSubmit
```

Comment:
probaly could just define the handleSubmit function here since you reference it directly in the JSX